### PR TITLE
document `features` for `MapMouseEvent` and `MapTouchEvent`

### PR DESCRIPTION
**Pull request by ansis**
`features` has been an undocumented property of `MapMouseEvent` and `MapTouchEvent`. This documents it.

 It is only set when `layerId` is specified for the listener. I'm not sure that having this property is ideal, but I don't think we can remove it at this point. It has existed for long enough (and has been used in docs examples) that we can't remove it without breaking things. I think the underlying logic for this property was: we need to query features internally when `layerId` is set to know if the event occurs on that layer so we might as well give them to the user. 

We could always set `features` but I don't think we should do this because of the performance cost.

Screen Shot 2021-07-07 at 11 28 24 AM

### CODE REVIEW

**Comment by colleenmcginnis** [comment]
Also fixes https://github.com/mapbox/mapbox-gl-js/issues/10151.

File: src/ui/events.js [hunk 1]
```diff
@@ -68,6 +68,30 @@ export class MapMouseEvent extends Event {
...
+     * map.on('click', 'myLayerId', function(e) {
+     *     console.log('There are', e.features.length, 'features at point', e.point');
+     * });
+     *
+     * @example
+     * // logging all features for all layers (without `e.features`)
+     * map.on('click', function(e) {
+     *     var features = map.queryRenderedFeatures(e.point);
+     *     console.log('There are', features.length, 'features at point', e.point');
+     * });
```
Comment (mourner) [inline src/ui/events.js:91]: Nit: if we're going the ES6 way anyway, we could do it in new examples straightaway:

```suggestion
     * map.on('click', 'myLayerId', (e) => {
     *     console.log(`There are ${e.features.length} features at point ${e.point}`);
     * });
     *
     * @example
     * // logging all features for all layers (without `e.features`)
     * map.on('click', (e) => {
     *     var features = map.queryRenderedFeatures(e.point);
     *     console.log(`There are ${e.features.length} features at point ${e.point}`);
     * });
```

**Review by mourner — APPROVED** [review]
This is great!

File: src/ui/events.js [hunk 2]
```diff
@@ -79,15 +79,15 @@ export class MapMouseEvent extends Event {
...
+     *     console.log(`There are ${e.features.length} features at point ${e.point}`);
      * });
      *
      * @example
      * // logging all features for all layers (without `e.features`)
-     * map.on('click', function(e) {
+     * map.on('click', (e) => {
      *     var features = map.queryRenderedFeatures(e.point);
-     *     console.log('There are', features.length, 'features at point', e.point');
+     *     console.log(`There are ${e.features.length} features at point ${e.point}`);
```
Comment (mourner) [inline src/ui/events.js:90]: Oopd, made a typo here (need to remove `e.` and also turn `var` into `const`.

**Pull request merged**